core/templates/components/entity-creation-services/exploration-creation.service.spec.ts

```diff
@@ -0,0 +1,224 @@
+        }
+      }
+    };
+  }
+}
+
+class MockWindowRef {
+  _window = {
+    location: {
+      _href: '',
+      get href(): string {
+        return this._href;
+      },
+      set href(val) {
+        this._href = val;
+      }
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+describe('ExplorationCreationService', () => {
+  let ecs: ExplorationCreationService;
+  let ecbas: MockExploratinoCreationBackendApiService;
+  let loaderService: LoaderService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  let urlInterpolationService: UrlInterpolationService;
+  let csrfTokenService: CsrfTokenService;
+  let alertsService: AlertsService;
+  let windowRef: MockWindowRef;
+  let ngbModal: NgbModal;
+
+  beforeEach(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: ExplorationCreationBackendApiService,
+          useClass: MockExploratinoCreationBackendApiService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        }
+      ]
+    });
+
+    ecs = TestBed.inject(ExplorationCreationService);
+    ecbas = (TestBed.inject(
+      ExplorationCreationBackendApiService) as unknown) as
+      MockExploratinoCreationBackendApiService;
+    loaderService = TestBed.inject(LoaderService);
+    siteAnalyticsService = TestBed.inject(SiteAnalyticsService);
+    urlInterpolationService = TestBed.inject(UrlInterpolationService);
+    csrfTokenService = TestBed.inject(CsrfTokenService);
+    alertsService = TestBed.inject(AlertsService);
+    ngbModal = TestBed.inject(NgbModal);
+  });
+
+  it('should not create a new exploration if another exploration' +
+    ' creation is in progress', () => {
+    spyOn(ecbas, 'registerNewExplorationAsync');
+    ecs.explorationCreationInProgress = true;
+
+    expect(ecs.createNewExploration()).toBe(undefined);
+    expect(ecbas.registerNewExplorationAsync).not.toHaveBeenCalled();
+  });
+
+  it('should show \'Creating exploration\' loading screen while' +
+    ' creating exploration', () => {
+    spyOn(ecbas, 'registerNewExplorationAsync').and.returnValue(
+      new Promise<ExplorationCreationResponse>((resolve, reject) => {}));
+    spyOn(loaderService, 'showLoadingScreen');
+    ecs.explorationCreationInProgress = false;
+
+    ecs.createNewExploration();
+
+    expect(loaderService.showLoadingScreen)
+      .toHaveBeenCalledWith('Creating exploration');
+    expect(ecbas.registerNewExplorationAsync).toHaveBeenCalled();
+  });
+
+  it('should create new exploration', fakeAsync(() => {
+    spyOn(siteAnalyticsService, 'registerCreateNewExplorationEvent');
+    spyOn(urlInterpolationService, 'interpolateUrl').and.returnValue(
+      '/url/to/exp1'
+    );
+
+    expect(ecs.explorationCreationInProgress).toBe(undefined);
+    expect(windowRef.nativeWindow.location.href).toBe('');
+
+    ecs.createNewExploration();
+    tick(150);
```

Comment:
is it expected to use `tick` for async function tests?

/cc @srijanreddy98

Reply:
Yes, it can be used - [reference](https://codecraft.tv/courses/angular/unit-testing/asynchronous/#_code_fakeasync_code_and_code_tick_code)

Reply:
@srijanreddy98 Can you please confirm whether it's fine to use tick here?

Reply:
Yes, it's correct to use tick here.